native/cocos/platform/BasePlatform.cpp

```diff
@@ -66,4 +74,14 @@ BasePlatform* BasePlatform::getPlatform() {
 #endif
     return &platform;
 }
-} // namespace cc
\ No newline at end of file
+
+BasePlatform* BasePlatform::getPlatform() {
+    if (_currentPlatform) {
+        return _currentPlatform;
+    }
+    createDefaultPlatform();
```

Comment:
Is `createDefaultPlatform ` needed?

Reply:
Yes, this is to work with developers using custom platforms.
At the template layer users can use:
```
class  CustomMacPlatform : public MacPlatform {
public:
     int run() override;
}
```
in the main function：
```
int main( ) {
     CustomMacPlatform platform;
     platform.init();
     return platform.start();
}
```

Reply:
It is a private static function, what's the relationship with developer's codes?

Reply:
If the user needs to extend the platform function, there is no need to create it internally